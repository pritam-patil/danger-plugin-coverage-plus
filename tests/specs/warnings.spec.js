import mockFs from 'mock-fs';

import coverage from '../../src';
import { setupEnv } from '../utils';

describe('Warnings', () => {
  beforeEach(setupEnv);

  afterEach(() => {
    mockFs.restore();
  });

  it('warns if no report detected', async () => {
    mockFs();

    Object.assign(danger, {
      git: {
        created_files: ['from-custom-report.js'],
        modified_files: [],
      },
    });

    await coverage();

    const [warning] = warn.mock.calls?.[0] || [];

    expect(warning).toMatch(/No coverage report.*/);
    expect(warning).toMatchSnapshot();
  });

  it('does not log any warnings on no report if disabled', async () => {
    mockFs();

    Object.assign(danger, {
      git: {
        created_files: ['from-custom-report.js'],
        modified_files: [],
      },
    });

    await coverage({ warnOnNoReport: false });

    const [warning] = warn.mock.calls?.[0] || [];

    expect(warning).toBeUndefined();
  });
});
