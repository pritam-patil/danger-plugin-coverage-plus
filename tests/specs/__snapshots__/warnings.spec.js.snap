// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Warnings warns if no report detected 1`] = `"No coverage report was detected. Please output a report in the \`clover.xml\` format before running danger"`;
